include/souper/Infer/DataflowPruning.h

```diff
@@ -0,0 +1,43 @@
+#include "llvm/ADT/APInt.h"
+#include "llvm/Support/KnownBits.h"
+#include "llvm/IR/ConstantRange.h"
+#include "souper/Inst/Inst.h"
+
+#include <unordered_map>
+
+namespace souper {
+namespace dataflow {
+
+struct EvalValue {
+  EvalValue(llvm::APInt Val_, bool Poison_, bool Unavailable_)
+    : Val(Val_), Poison(Poison_), Unavailable(Unavailable_) {}
+  EvalValue() : Unavailable(true) {}
+  llvm::APInt Val;
+  bool Poison;
```

Comment:
what about undef?

Reply:
Switched to llvm::Optional, no longer considering poison as a separate case.  
The current implementation will just not prune in case of poison or undef, which is sound.